src/record-error-archival/src/DatabaseClient.ts

```diff
@@ -0,0 +1,129 @@
+import { Client } from "pg"
+import { logger } from "shared"
+
+export type ArchivedErrorRecord = {
+  messageId: string
+  errorId: number
+  archivedAt: Date
+  archivedBy: string
+  archiveLogId: number
+}
+
+// Produces the string "$1, $2, $3..." for the given range
+const wherePlaceholderForRange = (range: number) => [...Array(range).keys()].map((x) => `$${x + 1}`).join(", ")
+
+export default class DatabaseClient {
+  private host: string
+
+  private user: string
+
+  private password: string
+
+  private database: string
+
+  private useSsl: boolean
+
+  private schema: string
+
+  private postgres: Client
+
+  private archiveGroupLimit: number
+
+  constructor(
+    host: string,
+    user: string,
+    password: string,
+    useSsl: boolean,
+    database: string,
+    schema: string,
+    archiveGroupLimit: number
+  ) {
+    this.host = host
+    this.user = user
+    this.password = password
+    this.useSsl = useSsl
+    this.database = database
+    this.schema = schema
+    this.archiveGroupLimit = archiveGroupLimit
+
+    this.postgres = new Client({
+      host: this.host,
+      user: this.user,
+      password: this.password,
+      ssl: this.useSsl,
+      database: this.database
+    })
+  }
+
+  async connect() {
+    await this.postgres.connect()
+  }
+
+  async disconnect() {
+    await this.postgres.end()
+  }
+
+  async fetchUnloggedArchivedErrors(): Promise<ArchivedErrorRecord[]> {
+    logger.debug("Fetching unlogged archived errors")
+
+    const errors: ArchivedErrorRecord[] = await this.postgres
```

Comment:
What happens if this errors? Should it have a .catch on the promise? Elsewhere we use a PromiseResult type (that we've made) to allow us to return the expected result or an Error object